### PR TITLE
feat(share: completions): add for 'pacaur'

### PR DESCRIPTION
**Pull request by actionless**

### CODE REVIEW

**Comment by actionless** [comment]
i copied it from `yaourt` one, but mb it could be some way to generalize the completions for `pacman` wrappers?
i've noticed what in previous versions of `fish` `yaourt` completion was reusing `pacman`s one

File: share/completions/pacaur.fish [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+complete -c $progname -s U -f -l upgrade -n $noopt -d 'Upgrade or add a local package'
+complete -c $progname -s F -f -l files -n $noopt -d 'Query the files database'
+complete -c $progname -s V -f -l version -d 'Display version and exit'
+complete -c $progname -s h -f -l help -d 'Display help'
+# pacaur operations
+complete -c $progname -n $noopt -s s    -l search      -d "(AUR) Search for packages"
+complete -c $progname -n $noopt -s i    -l info        -d "(AUR) Show info for packages"
+complete -c $progname -n $noopt -s d    -l download    -d "(AUR) Clone the packages' build files"
+# @TODO:
+#complete -c $progname -n __fish_contains_opt -s d download -s d -l download -d 'Download recursively'
```
Comment (actionless) [inline share/completions/pacaur.fish]: please help me with with one:
```
       -d, --download
           Clone the target(s) build files. Pass this option twice to fetch uninstalled dependencies (done
           recursively).
```

Comment (faho) [inline share/completions/pacaur.fish]: From the pacman completion: 

```fish
complete -c $progname -n "$sync; and not __fish_contains_opt -s u sysupgrade" -s u -l sysupgrade -d 'Upgrade all packages that are out of date'
complete -c $progname -n "$sync; and __fish_contains_opt -s u sysupgrade" -s u -l sysupgrade -d 'Also downgrade packages'
```

Unfortunately that seems to not generate `-Suu` properly, only `-Su -u` and the long variant work.

**Comment by faho** [comment]
The issue with wrapping here, IIRC, is the "has_opt" thing. That is of course different for wrappers that add options. Previously the pacman completion supplied a function `_fish_complete_pacman` or something like it, but that wasn't used for much and could be completely replaced with the `complete --wraps` calls in e.g. the pacman-color completions.

File: share/completions/pacaur.fish (hunk 1 above)
Comment (actionless) [inline share/completions/pacaur.fish]: thanks!

**Comment by actionless** [comment]
i think now it's ready for the review, only blocks starting with comment containing "pacaur" string are actually changed

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**